src/bsk201910/rollbacks.md

```diff
@@ -0,0 +1,407 @@
+[TODO: Nice intro]
+
+Even if you use techniques such as Rolling updates, Canary and Blue-green Deployments, there's still risk that your application doesn't work the way you expect.
+
+**When you introduce a change that breaks production, you should have a plan to roll back that change.**
+
+Kubernetes and `kubectl` offer a simple mechanism to roll back changes to resources such as Deployments.
+
+But before talking about rollbacks, you should learn an important detail about Deployments.
+
+You learned how Deployments are responsible for gradually rolling out new versions of your Pods without causing any downtime.
+
+You are also familiar with the fact that Kubernetes watches over the number of replicas in your deployment.
+
+_If you asked for 5 Pods but have only 4, Kubernetes creates one more._
+
+_If you asked for 4 Pods, but you have 5, Kubernetes deletes one of the running Pods._
+
+Since the replicas is a field in the Deployment, you might be tempted to conclude that is the Deployment's job to count the number of Pods and create or delete them.
+
+_This is not the case, unfortunately._
+
+**Deployments delegate counting Pods to another component called the ReplicaSet.**
+
+Every time you create a Deployment, the deployment creates a ReplicaSet and delegates creating (and deleting) the Pods.
+
+```slideshow
+{
+  "description": "Deployments don't create Pods. ReplicaSets do.",
+  "slides": [
+    {
+      "image": "assets/replicaset-2.svg",
+      "description": "Let's focus on a Deployment."
+    },
+    {
+      "image": "assets/replicaset-3.svg",
+      "description": "You might be tempted to think that Deployments are in charge of creating Pods."
+    },
+    {
+      "image": "assets/replicaset-4.svg",
+      "description": "The Deployment doesn't create Pods. Instead it creates another object called ReplicaSet."
+    },
+    {
+      "image": "assets/replicaset-5.svg",
+      "description": "The Deployment passes the spec (which includes the replicas) to the ReplicaSet."
```

Comment:
Hmm, you got me, we have `minReadySeconds`, but that is the last one. :)  You can check with `kubectl explain replicaset.spec`.

I think not talking about `minReadySeconds` is an OK simplification, because saying "passes the spec" is a much bigger lie, if you look at `kubectl explain deployment.spec` (that has quite some things in it).